x/token/msgs_test.go

```diff
@@ -741,3 +743,142 @@ func TestMsgRevokePermission(t *testing.T) {
+			&token.MsgGrantPermission{
+				ContractId: contractId,
+				From:       addrs[0].String(),
+				To:         addrs[1].String(),
+				Permission: token.LegacyPermissionMint.String(),
+			},
+			"/lbm.token.v1.MsgGrantPermission",
+			fmt.Sprintf("{\"account_number\":\"1\",\"chain_id\":\"foo\",\"fee\":{\"amount\":[],\"gas\":\"0\"},\"memo\":\"memo\",\"msgs\":[{\"type\":\"lbm-sdk/token/MsgGrantPermission\",\"value\":{\"contract_id\":\"deadbeef\",\"from\":\"%s\",\"permission\":\"mint\",\"to\":\"%s\"}}],\"sequence\":\"1\",\"timeout_height\":\"1\"}", addrs[0].String(), addrs[1].String()),
+		},
+		"MsgRevokePermission": {
+			&token.MsgRevokePermission{
+				ContractId: contractId,
+				From:       addrs[0].String(),
+				Permission: token.LegacyPermissionMint.String(),
+			},
+			"/lbm.token.v1.MsgRevokePermission",
+			fmt.Sprintf("{\"account_number\":\"1\",\"chain_id\":\"foo\",\"fee\":{\"amount\":[],\"gas\":\"0\"},\"memo\":\"memo\",\"msgs\":[{\"type\":\"lbm-sdk/token/MsgRevokePermission\",\"value\":{\"contract_id\":\"deadbeef\",\"from\":\"%s\",\"permission\":\"mint\"}}],\"sequence\":\"1\",\"timeout_height\":\"1\"}", addrs[0].String()),
+		},
+		"MsgMint": {
+			&token.MsgMint{
+				ContractId: contractId,
+				From:       addrs[0].String(),
+				To:         addrs[1].String(),
+				Amount:     sdk.NewInt(1000000),
+			},
+			"/lbm.token.v1.MsgMint",
+			fmt.Sprintf("{\"account_number\":\"1\",\"chain_id\":\"foo\",\"fee\":{\"amount\":[],\"gas\":\"0\"},\"memo\":\"memo\",\"msgs\":[{\"type\":\"lbm-sdk/MsgMint\",\"value\":{\"amount\":\"1000000\",\"contract_id\":\"deadbeef\",\"from\":\"%s\",\"to\":\"%s\"}}],\"sequence\":\"1\",\"timeout_height\":\"1\"}", addrs[0].String(), addrs[1].String()),
+		},
+		"MsgBurn": {
+			&token.MsgBurn{
+				ContractId: contractId,
+				From:       addrs[0].String(),
+				Amount:     sdk.Int{},
+			},
+			"/lbm.token.v1.MsgBurn",
+			fmt.Sprintf("{\"account_number\":\"1\",\"chain_id\":\"foo\",\"fee\":{\"amount\":[],\"gas\":\"0\"},\"memo\":\"memo\",\"msgs\":[{\"type\":\"lbm-sdk/MsgBurn\",\"value\":{\"amount\":\"0\",\"contract_id\":\"deadbeef\",\"from\":\"%s\"}}],\"sequence\":\"1\",\"timeout_height\":\"1\"}", addrs[0].String()),
+		},
+		"MsgBurnFrom": {
+			&token.MsgBurnFrom{
+				ContractId: contractId,
+				Proxy:      addrs[0].String(),
+				From:       addrs[1].String(),
+				Amount:     sdk.NewInt(1000000),
+			},
+			"/lbm.token.v1.MsgBurnFrom",
+			fmt.Sprintf("{\"account_number\":\"1\",\"chain_id\":\"foo\",\"fee\":{\"amount\":[],\"gas\":\"0\"},\"memo\":\"memo\",\"msgs\":[{\"type\":\"lbm-sdk/MsgBurnFrom\",\"value\":{\"amount\":\"1000000\",\"contract_id\":\"deadbeef\",\"from\":\"%s\",\"proxy\":\"%s\"}}],\"sequence\":\"1\",\"timeout_height\":\"1\"}", addrs[1].String(), addrs[0].String()),
+		},
+		"MsgModify": {
+			&token.MsgModify{
+				ContractId: contractId,
+				Owner:      addrs[0].String(),
+				Changes:    []token.Pair{token.Pair{Field: token.AttributeKeyName.String(), Value: "New test"}},
```

Comment:
> warning
redundant type from array, slice, or map composite literalsimplifycompositelit

`token.Pair` is redundant.